src/cloud-api-adaptor/azure/cmd/copy-image/main.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	armcompute "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+)
+
+func main() {
+	var (
+		subscriptionID        string
+		resourceGroup         string
+		location              string
+		userImageName         string
+		managedDiskName       string
+		galleryName           string
+		imageDefinitionName   string
+		imageVersionName      string
+		communityGalleryImage string
+		targetRegionsStr      string
+	)
+
+	flag.StringVar(&subscriptionID, "subscription-id", os.Getenv("AZURE_SUBSCRIPTION_ID"), "Azure subscription ID")
+	flag.StringVar(&resourceGroup, "resource-group", os.Getenv("AZURE_RESOURCE_GROUP"), "Resource group name")
+	flag.StringVar(&location, "location", os.Getenv("AZURE_REGION"), "Azure region")
+	flag.StringVar(&userImageName, "user-image-name", "from-community-gallery-user-image", "Name of the temporary managed image")
+	flag.StringVar(&managedDiskName, "managed-disk-name", "from-community-gallery", "Name of the temporary managed disk")
```

Comment:
this is an implementation detail and shouldn't be exposed. we probably want to generate temporary names.